### PR TITLE
.github/workflows: use gh to upload release artifacts

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Uses [`gh release upload`](https://cli.github.com/manual/gh_release_upload) for the sake of pipeline code golfing, if that's a thing.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -64,7 +64,7 @@ jobs:
           path: age-*
   upload:
     name: Upload release binaries
-    if: ${{ github.event_name == 'release' }}
+    if: github.event_name == 'release'
```
Comment (0x2b3bfa0) [inline .github/workflows/build.yml:67]: This suggestion deviates from the pull request topic

**Comment by FiloSottile** [comment]
Oh, thanks, this is definitely better. Why `GH_REPO` and why `GH_TOKEN` instead of `GITHUB_TOKEN` as documented in https://docs.github.com/en/actions/using-workflows/using-github-cli-in-workflows.

**Comment by 0x2b3bfa0** [comment]
> _Why `GH_REPO`_

Because the `upload` job does not use the [`checkout`](https://github.com/actions/checkout) action. Thus, when `gh` is run, it won't find a Git repository nor know where to upload the artifacts.

```console
$ git release upload ...
fatal: not a git repository (or any of the parent directories): .git
/usr/local/bin/git: exit status 128
```

The easiest ways of specifying the target repository are either specifying the `--repo` option or setting the [`GH_REPO`](https://cli.github.com/manual/gh_help_environment) environment variable.

> _Why [`GH_TOKEN`](https://cli.github.com/manual/gh_help_environment) instead of `GITHUB_TOKEN`_

Just for foolish consistency with the `GH_REPO` environment variable; feel free to use `GITHUB_TOKEN` if you deem it more appropriate. The same applies for `github.token` versus `secrets.GITHUB_TOKEN` to retrieve it.

**Comment by FiloSottile** [comment]
That all makes sense, thank you!

**Pull request merged**